staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -109,7 +111,22 @@ func New(
 	clusterName string,
 	featureGate featuregate.FeatureGate,
 ) (*Controller, error) {
-	broadcaster := record.NewBroadcaster()
+	return NewWithContext(context.Background(), cloud, kubeClient, serviceInformer, nodeInformer, clusterName, featureGate)
+}
+
+// New returns a new service controller to keep cloud provider service resources
+// (like load balancers) in sync with the registry.
+func NewWithContext(
+	ctx context.Context,
+	cloud cloudprovider.Interface,
+	kubeClient clientset.Interface,
+	serviceInformer coreinformers.ServiceInformer,
+	nodeInformer coreinformers.NodeInformer,
+	clusterName string,
+	featureGate featuregate.FeatureGate,
+) (*Controller, error) {
+
+	broadcaster := record.NewBroadcaster(record.WithContext(ctx))
```

Comment:
hmm , but the broadcaster has a shutdown method.

It does not seem right to pass a context to a constructor of an object that have a Run method that already receives the context, that is the context that should shutdown the broadcaster

Reply:
It wasn't possible to do it differently with `NewBroadcaster` without breaking that API.

Perhaps here constructing the broadcaster should get delayed until the `Run` method gets invoked? Then the controller `New` methods remain free of the context dependency.